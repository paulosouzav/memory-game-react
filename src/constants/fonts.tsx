import { css } from 'styled-components';
import Roboto from '../assets/fonts/Roboto-Regular.ttf';
import RobotoThin from '../assets/fonts/Roboto-Thin.ttf';
import RobotoLight from '../assets/fonts/Roboto-Light.ttf';
import RobotoBold from '../assets/fonts/Roboto-Bold.ttf';

export const FONTS = Object.freeze({
  REGULAR: 'Roboto',
  THIN: 'RobotoThin',
  LIGHT: 'RobotoLight',
  BOLD: 'RobotoBold',
});

interface FontFace {
  name: string;
  url: any;
  weight: string | number;
}

const FONT_FACES_LIST: Array<FontFace> = [
  {
    name: FONTS.REGULAR,
    url: Roboto,
    weight: 'normal',
  },
  {
    name: FONTS.THIN,
    url: RobotoThin,
    weight: 100,
  },
  {
    name: FONTS.LIGHT,
    url: RobotoLight,
    weight: 300,
  },
  {
    name: FONTS.BOLD,
    url: RobotoBold,
    weight: 'bold',
  },
];

export const FONT_FACES = css`
  ${FONT_FACES_LIST.map(
    ({ name, url, weight }: FontFace) => css`
      @font-face {
        font-family: ${name}, 'sans-serif';
        src: url(${url});
        font-weight: ${weight};
      }
    `
  )}
`;
