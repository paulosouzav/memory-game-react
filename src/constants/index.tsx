export { CARDS } from './cards';
export { COLORS } from './colors';
export { ROTATION } from './rotation';
export { ROUTES } from './routes';
export { BREAKPOINTS, MAX_WIDTHS } from './screen';
export { SHADOWS } from './shadows';
export { SIZES } from './sizes';
