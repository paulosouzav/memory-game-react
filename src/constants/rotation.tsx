import { css } from 'styled-components';

export const ROTATION = {
  half: css`
    transform: rotateY(180deg);
  `,
};
