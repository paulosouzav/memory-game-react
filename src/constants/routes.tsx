import { Game } from '../views';

interface route {
  id: string;
  path: string;
  description: string;
  Component: Function;
}

export const ROUTES: Array<route> = [
  {
    id: 'game',
    path: '/',
    description: 'Game',
    Component: Game,
  },
];
