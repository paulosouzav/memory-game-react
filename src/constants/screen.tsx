interface Indexable {
  [key: string]: number;
}

export const MAX_WIDTHS: Indexable = Object.freeze({
  sm: 540,
  md: 720,
  lg: 960,
  xl: 1140,
});

export const BREAKPOINTS: Indexable = Object.freeze({
  sm: 576,
  md: 758,
  lg: 992,
  xl: 1200,
});
