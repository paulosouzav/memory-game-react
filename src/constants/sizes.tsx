export const SIZES = Object.freeze({
  navHeight: '60px',
});
