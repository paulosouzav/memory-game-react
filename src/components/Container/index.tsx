import styled, { css } from 'styled-components';
import { MAX_WIDTHS, SIZES } from '../../constants';
import { mediaMin } from '../../utils';

// Modifiers should always be optional
// so always add the "?" (question mark) sign
interface Modifiers {
  fluid?: boolean;
  topSpaced?: boolean;
}

export const Container = styled.div`
  width: 100%;
  height: 100%;
  margin: 0 auto;
  padding-right: 15px;
  padding-left: 15px;
  
  ${mediaMin.sm`max-width: ${MAX_WIDTHS.sm}px`}
  ${mediaMin.md`max-width: ${MAX_WIDTHS.md}px`}
  ${mediaMin.lg`max-width: ${MAX_WIDTHS.lg}px`}
  ${mediaMin.xl`max-width: ${MAX_WIDTHS.xl}px`}

  ${({ fluid }: Modifiers) => fluid && FLUID}
  ${({ topSpaced }: Modifiers) => topSpaced && TOP_SPACED}
`;

const FLUID = css`
  max-width: 100% !important;
`;

const TOP_SPACED = css`
  margin-top: ${SIZES.navHeight};
`;
