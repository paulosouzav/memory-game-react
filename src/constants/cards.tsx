import Asteroid from '../assets/cards/002-asteroid.png';
import Earth from '../assets/cards/006-planet-earth.png';
import Mars from '../assets/cards/017-mars.png';
import Sun from '../assets/cards/033-sun.png';
import Saturn from '../assets/cards/028-saturn.png';
import Spaceship from '../assets/cards/014-space-ship.png';
import DefaultImage from '../assets/cards/default-card.png';
import { ICard } from '../components/Card';
import { IImage } from '../components/Image';

export const DEFAULT_CARD_IMAGE: IImage = {
  src: DefaultImage,
  alt: 'Default',
};

export const CARDS: Array<ICard> = [
  {
    id: 'asteroid',
    image: {
      src: Asteroid,
      alt: 'Asteroid',
    },
  },
  {
    id: 'earth',
    image: {
      src: Earth,
      alt: 'Earth',
    },
  },
  {
    id: 'mars',
    image: {
      src: Mars,
      alt: 'Mars',
    },
  },
  {
    id: 'saturn',
    image: {
      src: Saturn,
      alt: 'Saturn',
    },
  },
  {
    id: 'spaceship',
    image: {
      src: Spaceship,
      alt: 'Spaceship',
    },
  },
  {
    id: 'sun',
    image: {
      src: Sun,
      alt: 'Sun',
    },
  },
];
