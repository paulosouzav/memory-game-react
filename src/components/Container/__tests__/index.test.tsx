import React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';
import Container from '../';
import { SIZES } from '../../../constants/sizes';

describe('BaseCard', () => {
  it('should render without failing', () => {
    const wrapper = mount(<Container />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render fluid container', () => {
    const wrapper = mount(<Container fluid />);
    expect(wrapper).toHaveStyleRule('max-width', '100% !important');
  });

  it('should render top spaced container', () => {
    const wrapper = mount(<Container topSpaced />);
    expect(wrapper).toHaveStyleRule('margin-top', SIZES.navHeight);
  });
});
