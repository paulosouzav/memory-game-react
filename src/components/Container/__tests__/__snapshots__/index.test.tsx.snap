// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseCard should render without failing 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  margin: 0 auto;
  padding-right: 15px;
  padding-left: 15px;
}

@media (min-width:576px) {
  .c0 {
    max-width: 540px;
  }
}

@media (min-width:758px) {
  .c0 {
    max-width: 720px;
  }
}

@media (min-width:992px) {
  .c0 {
    max-width: 960px;
  }
}

@media (min-width:1200px) {
  .c0 {
    max-width: 1140px;
  }
}

<Container>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  width: 100%;
  height: 100%;
  margin: 0 auto;
  padding-right: 15px;
  padding-left: 15px;
  
  ",
            "
      @media (",
            "min",
            "-width: ",
            "576",
            "px) {
        ",
            "max-width: ",
            "540",
            "px",
            ";
      }
    ",
            "
  ",
            "
      @media (",
            "min",
            "-width: ",
            "758",
            "px) {
        ",
            "max-width: ",
            "720",
            "px",
            ";
      }
    ",
            "
  ",
            "
      @media (",
            "min",
            "-width: ",
            "992",
            "px) {
        ",
            "max-width: ",
            "960",
            "px",
            ";
      }
    ",
            "
  ",
            "
      @media (",
            "min",
            "-width: ",
            "1200",
            "px) {
        ",
            "max-width: ",
            "1140",
            "px",
            ";
      }
    ",
            "

  ",
            [Function],
            "
  ",
            [Function],
            "
",
          ],
        },
        "displayName": "Container",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="c0"
    />
  </StyledComponent>
</Container>
`;
